### PR TITLE
Paint can be used on Unions & Negations

### PR DESCRIPTION
**Pull request by AnonymousPrayers**
Added some code so that when a Union or Negation's color is changed, its UsePartColor property is set to true, so that the color actually changes. Works with history, too.

### CODE REVIEW

**Comment by GigsD4X** [comment]
Nice! :+1:

File: tools/Paint.lua [hunk 1]
```diff
@@ -83,6 +84,9 @@ Tools.Paint.startHistoryRecord = function ( self )
 			Selection:clear();
 			for _, Target in pairs( self.targets ) do
 				if Target then
+					if Target:IsA('PartOperation') then
+						Target.UsePartColor = true
```
Comment (GigsD4X) [inline tools/Paint.lua]: Change to `Target.UsePartColor = self.initial_color_states[Target];`

File: tools/Paint.lua [hunk 2]
```diff
@@ -83,6 +84,9 @@ Tools.Paint.startHistoryRecord = function ( self )
...
+					end
```
Comment (GigsD4X) [inline tools/Paint.lua]: Add semicolon

File: tools/Paint.lua [hunk 3]
```diff
@@ -92,6 +96,9 @@ Tools.Paint.startHistoryRecord = function ( self )
 	for _, Item in pairs( self.State.HistoryRecord.targets ) do
 		if Item then
 			self.State.HistoryRecord.initial_colors[Item] = Item.BrickColor;
+			if Item:IsA('PartOperation') then
+				self.State.HistoryRecord.initial_color_states[Item] = Item.UsePartColor
+			end
```
Comment (GigsD4X) [inline tools/Paint.lua]: Add semicolons

File: tools/Paint.lua [hunk 4]
```diff
@@ -128,6 +135,9 @@ Tools.Paint.Listeners.Button1Up = function ()
 		-- Paint all of the selected items `Tools.Paint.Options.Color`
 		if self.Options.Color then
 			for _, Item in pairs( Selection.Items ) do
+				if Item:IsA('PartOperation') then
+					Item.UsePartColor = true
+				end
```
Comment (GigsD4X) [inline tools/Paint.lua]: Add semicolons

File: tools/Paint.lua [hunk 5]
```diff
@@ -150,6 +160,9 @@ Tools.Paint.changeColor = function ( self, Color )
 
 		-- Then, we want to update the color of any items in the selection
 		for _, Item in pairs( Selection.Items ) do
+			if Item:IsA('PartOperation') then
+				Item.UsePartColor = true
+			end
```
Comment (GigsD4X) [inline tools/Paint.lua]: Add semicolons

**Comment by ThatChristianGuy** [comment]
Moved to #82

**Pull request closed without merging**